### PR TITLE
Finished Project

### PR DESCRIPTION
**Pull request by gdumani**
The following features were implemented in this PR

1.  Rockets page
      - Fetch rockets data
      - Reserve Rocket
      - Cancel reservation
2.  Missions page
     -  Fetch mission data
      - Join Mission
      - Leave Mission
3.   Profile page
     -  Display joined missions
      - Display reserved rockets

### CODE REVIEW

File: src/redux/missions/missions.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+eslint no-param-reassign: ["error", { "props": false }]
+--------------
+linter disable reason:
+According to the Redux Toolkit documentation state parameter should be reassigned.
+more info in: https://redux-toolkit.js.org/api/createSlice
+*/
```
Comment (Omar-Muhamad) [inline src/redux/missions/missions.js:7]: - [ ] Please remove all commented (unused) code to make your js modules more clean and readable.

File: src/redux/rockets/rockets.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          };
+        }
+        return { ...rocket };
+      });
+      return newState;
+    }
+    default:
+      return state;
+  }
+};
```
Comment (Omar-Muhamad) [inline src/redux/rockets/rockets.js:52]: - [ ] [Optional] [Highly recommended] Please use `createSlice` and `createAsyncThunk` to create rocket redux slice and thunk actions like you did in missions redux slice.

**Review by Omar-Muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @gdumani and @Juank628,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- [ ] Please check the inline comments below.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi team 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**